src/main/java/mods/railcraft/client/sounds/RCSoundHandler.java

```diff
@@ -59,18 +56,19 @@ public void onPlaySound(PlaySoundEvent event) {
                         newName = sound.getBreakSound();
                     else if (soundName.contains("place"))
                         newName = sound.getPlaceSound();
-                    event.result = new PositionedSoundRecord(new ResourceLocation(newName), event.sound.getVolume(), event.sound.getPitch() * sound.getPitch(), x, y, z);
+                    event.result = new PositionedSoundRecord(new ResourceLocation(newName), event.sound.getVolume(), event.sound.getPitch() * sound.getFrequency(), x, y, z);
                 }
             }
         }
     }
 
-    private SoundType getBlockSound(World world, int x, int y, int z) {
-        Block block = world.getBlockState(new BlockPos(x, y, z)).getBlock();
+    private SoundType getBlockSound(World world, BlockPos pos) {
+        IBlockState state = world.getBlockState(pos);
+        Block block = state.getBlock();
         if (block instanceof IBlockSoundProvider)
-            return ((IBlockSoundProvider) block).getSound(world, x, y, z);
+            return ((IBlockSoundProvider) block).getSound(world, pos);
         else {
-            int meta = world.getBlockMetadata(x, y, z);
+            int meta = block.getMetaFromState(state);
```

Comment:
All block meta comparisons -> BlockState comparisons? Or just change the sound handler?